packages/features/auth/lib/next-auth-options.ts

```diff
@@ -560,7 +560,7 @@ export const AUTH_OPTIONS: AuthOptions = {
                 console.error("Error while linking account of already existing user");
               }
             }
-            if (existingUser.twoFactorEnabled) {
+            if (existingUser.twoFactorEnabled && existingUser.identityProvider === IdentityProvider.CAL) {
```

Comment:
This missing check was not letting a user with a different provider than CAL to set the password as null and changing the provider with the new one which is later in the logic